packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -97,6 +97,8 @@ export class GarbageCollector implements IGarbageCollector {
 
 	// Keeps track of the GC state from the last run.
 	private gcDataFromLastRun: IGarbageCollectionData | undefined;
+	/** Nodes whose state in unreferencedNodesState changed between runs */
+	private readonly updatedNodesSinceLastRun: Set<string> = new Set();
```

Comment:
This looks like a duplicate of `newReferencesSinceLastRun`. The difference being it only includes the outbound routes instead of the references. Can we use only one of these instead of maintaining two states?

Reply:
I'll take a look. I was a little bit wondering that too, but my initial thought was this was is more explicit (but... the names don't explain the difference)

Reply:
Ok the difference is that I need to know specifically what _was unreferenced_, and `newReferencesSinceLastRun` only says what _is now referenced_ (regardless of whether it was unreferenced before).

But you made me realize I missed the key step of clearing this state in Post-GC steps.  Considering how to make it clearer and less error-prone.